client/comms/wsconn.go

```diff
@@ -402,7 +401,13 @@ func (conn *wsConn) readRaw(ctx context.Context) {
 			conn.handleReadError(err)
 			return
 		}
+
 		conn.cfg.RawHandler(msgBytes)
+
+		err = ws.SetReadDeadline(time.Now().Add(conn.cfg.PingWait))
+		if err != nil {
+			conn.log.Errorf("set read deadline failed: %v", err)
+		}
```

Comment:
This is handled by our `SetPingHandler` function. It shouldn't be necessary anywhere else. What are you seeing?

Reply:
I see your note about pings/heartbeats now. This solution is a little hacky, and doesn't really capture the intention of ping/pong or the heartbeat channel. Wouldn't it be better to reset the read deadline from **coinbase.go** when the heartbeat is received.

Reply:
Should `PingWait` just be renamed to `ReadWait`? Is there a reason to specifically wait for ping messages? The reason for pings is to confirm that the connection is alive, and if a non-ping message comes through, it confirms it as well.